test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -259,7 +259,7 @@
 (deftest ^:integration static-file-content-endpoint-test-no-code-content-command
   (logging/with-test-logging
    (testing "the /static_file_content endpoint errors if :code-content-command is not set"
-     (bootstrap/with-puppetserver-running
+     (bootstrap/with-puppetserver-running-with-mock-jrubies
```

Comment:
So I think I'm good with using mocks in all the places that you did so in this namespace.  However, there are a couple of them that are confusing because they appear to be making a call to the catalog endpoint, which in my head would *definitely* need the jrubies to be real.  It works here though because the code-id check stuff happens in clojure before we would hand off the request, and fails.

I'm wondering if it would be worthwhile to add a comment at the beginning of this namespace explaining why it's OK?  Or if maybe we should have some general pattern around explaining why it's safe to use mocks on the various tests where we do it?

This may sound crazy (and it probably is) but I almost wonder if we should have something like a docstring argument to these macros, that just explains why mocking is safe for the test at hand.  Might save some confusion in the future?  Might also be total overkill?